core/modules.rs

```diff
@@ -579,11 +585,8 @@ impl ModuleMap {
     let imports = self.get_children(module_id).unwrap().clone();
 
     for module_specifier in imports {
-      let is_registered = self.is_registered(&module_specifier);
-      if !is_registered {
-        load
-          .add_import(module_specifier.to_owned(), referrer_specifier.clone());
-      }
+      load
+        .visit_import(module_specifier.to_owned(), referrer_specifier.clone());
```

Comment:
I'm afraid this will cause multiple loads of the same module, which might lead to race conditions